### PR TITLE
Adding SequenceToSequence guide

### PR DESCRIPTION
**Pull request by akhilnegi**
Added Guide for arcgis.learn's SequenceToSequence module.

### CODE REVIEW

File: guide/14-deep-learning/how-sequencetosequence-works.ipynb [hunk 1]
```diff
@@ -0,0 +1,859 @@
+{
```
Comment (anurag89sharma) [inline guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <ol><li>Add space between length and ( in the sentence <strong><u>output sequence of any length(independent</u></strong></li><li><s>Few such tasks are: </s> Sequence translation includes tasks such as machine translation, summarization, question answering, etc. (include relevant links from above)</li><li>Provide image description similar to - &lt;center&gt;Figure1: High level view of a transformer&lt;/a&gt;&lt;/center&gt;</li><li><s>Transformers library gives</s> Either use <strong>Hugging Face Transformers library provides</strong> or <strong>This library provides</strong></li><li><s>consists of an Encoder block and a decoder block</s> - consists of an <strong>encoder block</strong> and a <strong>decoder block</strong></li><li><s>Many of the latest transformer-based architectures like Bert, RoBerta, AlBert utilizes</s> - Many of the latest transformer  based architectures like BERT, RoBERTa, ALBERT utilizes</li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/869/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (anurag89sharma) [inline guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <ol><li>is the <s>the</s> input text column</li><li><code>std_address</code> <s>column</s> is the output text column, which</li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/869/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (anurag89sharma) [inline guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <ol><li><strong><s>"sequence_translation"</s></strong><s>&nbsp;for SequenceToSequence model</s> - Available choice for <code>SequenceToSequence</code> model is <strong>"sequence_translation"</strong></li><li>Content of LICENSE has to be changed. The license file should contain attribution for United states only and not for other countries. This license file contain attribution for house addresses for 30 countries (it was created specifically for sample notebook for TextClassifier)</li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/869/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (anurag89sharma) [inline guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p>to computer vision models in <code>arcgis.learn</code>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/869/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (anurag89sharma) [inline guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p><code>SequenceToSequence</code> class to see the <s>available </s>supported backbones</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/869/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (anurag89sharma) [inline guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p><s>method accepts one of the </s><code><s>supported_backbones</s></code> - method accepts a backbone name (one of the values returned by call to <code>SequenceToSequence.supported_backbones</code> ) as an argument </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/869/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (anurag89sharma) [inline guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p>The link above is to generic and contain links to all the backbone/architectures available in the Hugging Face Transformer library. See if we can provide link that contains models for T5, Marian and Bart architectures.  See if we can use this link https://huggingface.co/models?filter=translation </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/869/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (anurag89sharma) [inline guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p>fix - method of the <code>TextClassifier</code> class, should be changed to <code>SequenceToSequence</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/869/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

Comment (anurag89sharma) [inline guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p>method of the <code><s>TextClassifier</s></code>  <code>SequenceToSequence</code> class</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/869/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

Comment (anurag89sharma) [inline guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p>add space between accuracy &amp; [3] and bleu &amp; [4]</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/869/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Comment (anurag89sharma) [inline guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p>method of the <code><s>TextClassifier</s></code>  <code>SequenceToSequence</code> class</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/869/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

Comment (anurag89sharma) [inline guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p>add space between accuracy &amp; [3] and bleu &amp; [4]</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/869/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Comment (akhilnegi) [reply guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p>Need discussion.</p>

Comment (akhilnegi) [reply guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p>fixed</p>

Comment (akhilnegi) [reply guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p>fixed</p>

Comment (akhilnegi) [reply guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p>fixed</p>

Comment (akhilnegi) [reply guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p>fixed</p>

Comment (akhilnegi) [reply guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p>fixed</p>

Comment (akhilnegi) [reply guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p>UPDATED:</p><p> </p><p>Users can choose a supported backbone and a pretrained model based on the task the model is to be trained on. It is preferable to choose a model that was trained on a similar task and data.</p><p> </p><p>For instance, if a user is training a model for summarizing English text. It would be most appropriate to choose&nbsp;<code>T5</code>&nbsp;architecture and&nbsp;<code>t5-base-finetuned-summarize-news</code>&nbsp;model, as this model was trained for summarization downstream task.</p><p>One can search for available pretrained models in the&nbsp;<a href="https://huggingface.co/models" target="_blank">HuggingFace model zoo.</a></p>

Comment (akhilnegi) [reply guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p>please re-review the updated text once. Thanks!</p>

Comment (akhilnegi) [reply guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p>please re-review the updated text once. Thanks!</p>

Comment (akhilnegi) [reply guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p>fixed</p>

Comment (akhilnegi) [reply guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p>fixed</p>

Comment (akhilnegi) [reply guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p>fixed</p>

Comment (akhilnegi) [reply guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: <p>fixed</p>

Comment (akhilnegi) [reply guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: Resolved post discussion without any modification.

Comment (akhilnegi) [reply guide/14-deep-learning/how-sequencetosequence-works.ipynb:1]: modified as per the discussion.

**Comment by anurag89sharma** [comment]
Looks fine to me.

**Review by anurag89sharma — APPROVED** [review]

**Comment by mohi9282** [comment]
@akhilnegi Could you please add a title to the notebook right before TOC? Also, for the reference section - please list complete web addresses.

**Comment by mohi9282** [comment]
@akhilnegi any updates on the requested changes? We need to finalize the PRs today for release 1.8.4.

cc: @AtmaMani @priyankatuteja
